Common/src/geometry_structure.cpp

```diff
@@ -9387,1429 +9384,1580 @@ void CPhysicalGeometry::Read_CGNS_Format_Parallel(CConfig *config, string val_me
    only handle one database. ---*/
   
   if ( nbases > 1 ) {
-    SU2_MPI::Error("CGNS reader currently incapable of handling more than 1 database.", CURRENT_FUNCTION);
+    SU2_MPI::Error("CGNS reader currently can only handle 1 database.",
+                   CURRENT_FUNCTION);
   }
   
   /*--- Read the databases. Note that the CGNS indexing starts at 1. ---*/
   
-  for (int i = 1; i <= nbases; i++) {
+  if (cg_base_read(fn, iBase, basename, &cell_dim, &phys_dim)) cg_error_exit();
+  
+  /*--- Get the number of zones for this base. ---*/
+  
+  if (cg_nzones(fn, iBase, &nzones)) cg_error_exit();
+  if (rank == MASTER_NODE) {
+    cout << "Database " << iBase << ", " << basename << ": " << nzones;
+    cout << " zone(s), cell dimension of " << cell_dim << ", physical ";
+    cout << "dimension of " << phys_dim << "." << endl;
+  }
+  
+  /*--- Check if there is more than one zone. Throw an
+   error if the number of zones does not match the expected number. ---*/
+  
+  if (nzones != val_nZone) {
```

Comment:
Just to clarify: it is possible to use CGNS for multizone problems right? Is it also possible to have separate mesh files for every zone ?

Reply:
Right now all zones are assumed to be in the same file (and it does not impact performance).
To be able to specify zones with separate mesh file would be interesting. For this purpose, it would be nice to improve the interface of CGNS reader. Indeed, we should be able to select the CGNS base (correponding to solver type fluid, structure, thermal ...) and the zone in a CGNS file that we want to read. We can also rely on CGNS Family to select groups of CGNS zone to load. I fully agree with your idea to put the mesh reader in a separate file.

Reply:
As @MicK7 mentions, if all zones are in one file, it should work, and I can also confirm that separate files with a single zone each should also work for MZ problems. We should consider improving the reader to take more advantage of the CGNS standard (there are a lot of features we don't use), and when we do, we should try to match the layout that the major mesh packages use when exporting meshes for our kind of multizone / multiphysics problems.

I would like to divide up the geometry files soon in a second pass, at which point, I can move the mesh readers into their own classes potentially (like the output writers).